GeoLib/Polygon.cpp

```diff
@@ -417,27 +444,25 @@ void Polygon::splitPolygonAtPoint (std::list<GeoLib::Polygon*>::iterator polygon
 				std::swap (idx0, idx1);
 
 			// create two closed polylines
-			GeoLib::Polygon* polygon0 (new GeoLib::Polygon(*(*polygon_it)));
+			GeoLib::Polyline polyline0{*(*polygon_it)};
 			for (std::size_t k(0); k <= idx0; k++)
-				polygon0->addPoint ((*polygon_it)->getPointID (k));
-			for (std::size_t k(idx1 + 1); k < (*polygon_it)->getNumberOfPoints(); k++)
-				polygon0->addPoint ((*polygon_it)->getPointID (k));
-			polygon0->initialise();
+				polyline0.addPoint((*polygon_it)->getPointID(k));
+			for (std::size_t k(idx1 + 1);
```

Comment:
Declaration shadows a local variable `k`; see line 402. Maybe not a bug but could be nicer.